src/main/java/com/google/cloud/tools/maven/AppEngineStager.java

```diff
@@ -16,25 +16,20 @@
 
 package com.google.cloud.tools.maven;
 
-import java.nio.file.Files;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 
 public interface AppEngineStager {
+
   class Factory {
     static AppEngineStager newStager(StageMojo config) {
-      boolean isStandardStaging =
-          Files.exists(
-              config.sourceDirectory.toPath().resolve("WEB-INF").resolve("appengine-web.xml"));
-      if (isStandardStaging) {
-        return new AppEngineStandardStager(config);
-      } else {
-        return new AppEngineFlexibleStager(config);
-      }
+      return config.isStandardStaging()
+          ? new AppEngineStandardStager()
+          : new AppEngineFlexibleStager();
     }
   }
 
-  void stage() throws MojoExecutionException, MojoFailureException;
+  void stage(StageMojo stageMojo) throws MojoExecutionException, MojoFailureException;
 
-  void configureAppEngineDirectory();
+  void setAppEngineDirectory(StageMojo stageMojo);
```

Comment:
Maybe this method should be named something different, since it's not a typical setter.

Reply:
I took another look at this, turns out we don't need this at all. appengineDirectory is a flex only parameter and we can force it to be `src/main/appengine`.